Consistent javascript code style

markspanbroek opened the pull request:
I really like this Truffle box. Thanks for creating it!
I tend to use a linter to check for consistent code style, and it signalled a number of inconsistencies. Semicolons are mostly not used, but in some places they are. Sometimes single quotes are used for strings, sometimes double quotes. This pull request changes the formatting to comply with StandardJS (https://standardjs.com). I hope you like it.

adrianmcli commented:
Hey, thanks for the PR! I really appreciate you taking the time to do this, but I specifically made the decision not to include a linter because I wanted to leave that option up to the user. In my haste to do so, I probably left a few errors here and there, so thanks for pointing it out.

To be clear, I pretty much never do Javascript development without a linter. But for this Truffle Box, I don't think it would be a good idea to force linting upon the user. I hope this doesn't come across as off-putting, but I do welcome a consistent style. I'll be leaving some comments in your code, and if you're happy to meet me halfway, I'd be happy to merge this in.

There should probably be a linting solution for the development of this Truffle Box though, and then we can use the `truffle-box.json` to ignore it. I'll make an issue for this.

adrianmcli commented on client/lib/withWeb3.js:
```diff
@@ -1,11 +1,12 @@
+/* global alert */
```
Let's take this out because I don't want to enforce linting on the end user.

adrianmcli commented on client/package.json:
```diff
@@ -15,5 +15,8 @@
     "web3": "^1.0.0-beta.26"
+  },
+  "standard": {
+    "parser": "babel-eslint"
```
Let's take this out because I don't want to enforce linting on the end user.

adrianmcli commented on client/pages/dapp.js:
```diff
@@ -1,3 +1,4 @@
+/* global alert */
```
Let's take this out because I don't want to enforce linting on the end user.

adrianmcli commented on client/pages/dapp.js:
```diff
@@ -18,19 +19,18 @@ class Dapp extends React.Component {
   }
 
-  render() {
-    const { web3, accounts, contract } = this.props
```
I actually want to leave this one in here, because I want to make it clear that all three props are automatically injected. But we probably should actually use these objects in some manner.

Any ideas?

adrianmcli commented on migrations/1_initial_migration.js:
```diff
@@ -1,5 +1,6 @@
-const Migrations = artifacts.require("./Migrations.sol");
+/* global artifacts */
```
Let's take this out because I don't want to enforce linting on the end user.

adrianmcli commented on migrations/2_deploy_contracts.js:
```diff
@@ -1,5 +1,6 @@
-const SimpleStorage = artifacts.require("./SimpleStorage.sol");
+/* global artifacts */
```
Let's take this out because I don't want to enforce linting on the end user.

adrianmcli commented on test/simplestorage.js:
```diff
@@ -1,17 +1,17 @@
-const SimpleStorage = artifacts.require("./SimpleStorage.sol");
+/* eslint-env mocha */
+/* global artifacts contract assert */
```
Let's take these two lines out because I don't want to enforce linting on the end user.

markspanbroek commented:
That sounds sensible, I agree that it's better to not enforce linting on the user. I'll make some changes and update the pull request.

adrianmcli commented:
Thanks! looking forward to it 😄

markspanbroek commented:
I moved the linter configuration to the root package.json as specified in #11. I also added globals and envs there, removing the need for the `/* global */` and `/* eslint-env */` comments.

Linting can be done by running `npm run lint`.

I couldn't think of a good example of using web3, accounts and contract in the `render()` method, so instead I added a comment explaining how to get these variables.

adrianmcli commented:
Thanks for the great work, @markspanbroek!
